contracts/mainnet/connectors/pooltogether/main.sol

```diff
@@ -0,0 +1,203 @@
+        uint _amount = getUint(getId, amount);
+
+        PrizePoolInterface prizePoolContract = PrizePoolInterface(prizePool);
+
+        prizePoolContract.withdrawInstantlyFrom(from, _amount, controlledToken, maximumExitFee);
+
+        setUint(setId, _amount);
+
+        _eventName = "LogWithdrawInstantlyFrom(address,address,uint256,address,uint256,uint256,uint256)";
+        _eventParam = abi.encode(address(prizePool), address(from), _amount, address(controlledToken), maximumExitFee, getId, setId);
+    }
+
+    /**
+     * @dev Claim token from a Token Faucet
+     * @notice Claim token from a Token Faucet
+     * @param tokenFaucet TokenFaucet address
+     * @param user The user to claim tokens for
+     * @param setId Set claimed amount at this ID in `InstaMemory` Contract.
+    */
+    function claim (
+        address tokenFaucet,
+        address user,
+        uint256 setId
+    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
+        TokenFaucetInterface tokenFaucetContract = TokenFaucetInterface(tokenFaucet);
+
+        uint256 claimed = tokenFaucetContract.claim(user);
+
+        setUint(setId, claimed);
+
+        _eventName = "LogClaim(address,address, uint256, uint256)";
+        _eventParam = abi.encode(address(tokenFaucet), address(user), claimed, setId);
+    }
+
+    /**
+     * @dev Runs claim on all passed comptrollers for a user.
+     * @notice Runs claim on all passed comptrollers for a user.
+     * @param tokenFaucetProxyFactory The TokenFaucetProxyFactory address
+     * @param user The user to claim tokens for
+     * @param tokenFaucets The tokenFaucets to call claim on.
+    */
+    function claimAll (
+        address tokenFaucetProxyFactory,
+        address user,
+        TokenFaucetInterface[] calldata tokenFaucets
+    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
+        TokenFaucetProxyFactoryInterface tokenFaucetProxyFactoryContract = TokenFaucetProxyFactoryInterface(tokenFaucetProxyFactory);
+
+        tokenFaucetProxyFactoryContract.claimAll(user, tokenFaucets);
+
+        _eventName = "LogClaimAll(address,address,TokenFaucetInterface[])";
+        _eventParam = abi.encode(address(tokenFaucetProxyFactory), address(user), tokenFaucets);
+    }
+
+    /**
+     * @dev Deposit into Pod
+     * @notice Deposit assets into the Pod in exchange for share tokens
+     * @param prizePool PrizePool address to deposit to
+     * @param pod Pod address to deposit to
+     * @param to The address that shall receive the Pod shares
+     * @param tokenAmount The amount of tokens to deposit.  These are the same tokens used to deposit into the underlying prize pool.
+     * @param getId Get token amount at this ID from `InstaMemory` Contract.
+     * @param setId Set token amount at this ID in `InstaMemory` Contract.
+    */
+    function depositToPod(
+        address prizePool,
```

Comment:
@eccheung4 I think we can pass token address instead of prizePool address? It will save gas?

Reply:
Yes! You are right. I've made the change.